internal/dataplane/parser/translate_httproute.go

```diff
@@ -562,3 +591,133 @@ func httpBackendRefsToBackendRefs(httpBackendRef []gatewayv1beta1.HTTPBackendRef
 	}
 	return backendRefs
 }
+
+// ---------------------
+// POC methods
+// ---------------------
+
+func splitHTTPRouteRules(httpRoutes []*gatewayv1beta1.HTTPRoute) []gatewayv1beta1.HTTPRoute {
+	newHTTPRoutes := []gatewayv1beta1.HTTPRoute{}
+	hostnamedRoutes := []gatewayv1beta1.HTTPRoute{}
+	for _, route := range httpRoutes {
+		if len(route.Spec.Hostnames) == 0 {
+			hostnamedRoutes = append(hostnamedRoutes, *route)
+			continue
+		}
+		for _, hostname := range route.Spec.Hostnames {
+			hostnamedRoute := route.DeepCopy()
+			hostnamedRoute.Spec.Hostnames = []gatewayv1beta1.Hostname{hostname}
+			hostnamedRoutes = append(hostnamedRoutes, *hostnamedRoute)
+		}
+	}
+	for _, route := range hostnamedRoutes {
+		for _, rule := range route.Spec.Rules {
+			for _, match := range rule.Matches {
+				newRoute := route.DeepCopy()
+				newRoute.Spec.Rules = []gatewayv1beta1.HTTPRouteRule{
+					{
+						Matches: []gatewayv1beta1.HTTPRouteMatch{
+							match,
+						},
+						BackendRefs: rule.BackendRefs,
+						Filters:     rule.Filters,
+					},
+				}
+				newHTTPRoutes = append(newHTTPRoutes, *newRoute)
+			}
+		}
+	}
+
+	return newHTTPRoutes
+}
+
+func sortHTTPRoutes(routes []gatewayv1beta1.HTTPRoute) {
+	sort.SliceStable(routes, func(i, j int) bool {
+		hasHostnamei := hasHostnames(routes[i])
+		hasHostnamej := hasHostnames(routes[j])
+
+		matchi := routes[i].Spec.Rules[0].Matches[0]
```

Comment:
In my comprehension, the sort should run through ALL matches but not only for `HTTPRoutes`. For example, if two `HTTPRoute`s has the following spec:

```
rules:
  - matches:
    - path:  
      value: "/a/b/c"
      type: PathPrefix
    - path:
      value: "/a"
      type: PathPrefix
```
and 
```
rules:
  - matches:
    - path:
      value: "/a/b"
      type: PathPrefix
```
The priority of there matches should be interleaved, `/a/b/c` > `/a/b` >`/a`.

Reply:
Here is your example: https://github.com/Kong/kubernetes-ingress-controller/pull/4020/commits/2bc952a0c76700fc8bd9a6b8e21d3197b4c5f208